### PR TITLE
CDAP-15848. Config validations

### PR DESCRIPTION
**Pull request by aonischuk**
Validate input schema not null and input field is present and of correct type

### CODE REVIEW

**Comment by aonischuk** [comment]
@CuriousVini can you please review?

File: src/main/java/io/cdap/plugin/pdf/PDFExtractorConfig.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    return sourceFieldName;
+  }
+
+  public Boolean getContinueOnError() {
+    return continueOnError;
+  }
+
+  public void validate(FailureCollector failureCollector, Schema inputSchema) {
+    if (inputSchema == null) {
+      failureCollector.addFailure("Input schema should not be empty", null);
```
Comment (CuriousVini) [inline src/main/java/io/cdap/plugin/pdf/PDFExtractorConfig.java]: nit: period at the end

File: src/main/java/io/cdap/plugin/pdf/PDFExtractorConfig.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      return;
+    }
+
+    Schema.Field contentField = inputSchema.getField(sourceFieldName);
+    if (contentField == null) {
+      failureCollector.addFailure(String.format("Field '%s' is not present in input schema.", sourceFieldName),
+                                  null).withConfigProperty(PROPERTY_SOURCE_FIELD_NAME)
+        .withInputSchemaField(PROPERTY_SOURCE_FIELD_NAME, null);
```
Comment (CuriousVini) [inline src/main/java/io/cdap/plugin/pdf/PDFExtractorConfig.java]: nit: `.withInputSchemaField(PROPERTY_SOURCE_FIELD_NAME)`

File: src/main/java/io/cdap/plugin/pdf/PDFExtractorConfig.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+  public void validate(FailureCollector failureCollector, Schema inputSchema) {
+    if (inputSchema == null) {
+      failureCollector.addFailure("Input schema should not be empty", null);
+      return;
+    }
+
+    Schema.Field contentField = inputSchema.getField(sourceFieldName);
+    if (contentField == null) {
+      failureCollector.addFailure(String.format("Field '%s' is not present in input schema.", sourceFieldName),
```
Comment (CuriousVini) [inline src/main/java/io/cdap/plugin/pdf/PDFExtractorConfig.java]: error: Field '%s' must be present in the input schema.

File: src/main/java/io/cdap/plugin/pdf/PDFExtractorConfig.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        .withInputSchemaField(PROPERTY_SOURCE_FIELD_NAME, null);
+    } else {
+      Schema contentFieldSchema = contentField.getSchema();
+
+      if (contentFieldSchema.isNullable()) {
+        contentFieldSchema = contentFieldSchema.getNonNullable();
+      }
+
+      if (contentFieldSchema.getLogicalType() != null || contentFieldSchema.getType() != Schema.Type.BYTES) {
+        failureCollector.addFailure(String.format("Field '%s' must be of type 'bytes' but is of type '%s'.",
```
Comment (CuriousVini) [inline src/main/java/io/cdap/plugin/pdf/PDFExtractorConfig.java]: error: Field '%s' is of unexpected type '%s'.
corrective action: Supported type is bytes.

**Review by CuriousVini — APPROVED** [review]
minor comments. Please address them and squash

**Comment by aonischuk** [comment]
@CuriousVini thanks for the review. Made the changes, can you please check/merge?

**Pull request merged**